options/posix/generic/semaphore-stubs.cpp

```diff
@@ -98,7 +98,20 @@ int sem_unlink(const char *) {
 	__builtin_unreachable();
 }
 
-int sem_trywait(sem_t *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int sem_trywait(sem_t *sem) {
```

Comment:
This impl looks fine, but I feel like it could be reordered like so:
```cpp
while (true) {
    auto state = load(ctr);

    if (!(state & waiters_mask)) return EAGAIN;

    auto desired = state - 1;
    if (cmpxchg(ctr, state, desired)) return 0;
}